app/services/exporters/contracts.rb

```diff
@@ -29,18 +29,41 @@ def call
 
   attr_reader :cohort
 
-  def query
-    ContractTemplate
+  def attribute(field, record)
+    case field
+    when "monthly_service_fee"
+      record.attributes[field] || 0
+    else
+      record.attributes[field]
+    end
+  end
+
+  def contract_templates
+    statements_lead_provider_courses = Statement
       .joins(contracts: [{ statement: :lead_provider }, :course])
-      .where(statements: { cohort: })
-      .select(
-        "lead_providers.name as lead_provider_name",
-        "courses.identifier as course_identifier",
-        :recruitment_target,
-        :per_participant,
-        :service_fee_installments,
-        :special_course,
-        :monthly_service_fee,
-      ).distinct
+      .where(cohort:, output_fee: true)
+      .group(:lead_provider_id, :course_id)
+      .order(:lead_provider_id, :course_id)
+      .count
+
+    statements_lead_provider_courses.map do |values|
+      lead_provider_id, course_id = values[0]
+
+      ContractTemplate
+        .joins(contracts: [{ statement: :lead_provider }, :course])
+        .where(statements: { lead_provider_id: }, contracts: { course_id: })
+        .where("MAKE_DATE(statements.year, statements.month, 1) <= CURRENT_DATE")
+        .order("statements.year desc", "statements.month desc")
+        .limit(1)
+        .select(
+          "lead_providers.name as lead_provider_name",
+          "courses.identifier as course_identifier",
+          :recruitment_target,
+          :per_participant,
+          :service_fee_installments,
+          :special_course,
+          :monthly_service_fee,
+        ).first
```

Comment:
overall, this is way less efficient than before,
but I couldn't figure out a way of putting all of this into one SQL query.